apps/livreGUI/transferFunctionEditor/TransferFunctionEditor.h

```diff
@@ -52,28 +55,35 @@ class TransferFunctionEditor: public QWidget
 signals:
     void gradientStopsChanged( const QGradientStops& stops );
     void transferFunctionChanged();
+    void histogramChanged();
 
 private Q_SLOTS:
 
     void _clear();
     void _load();
     void _save();
     void _setDefault();
-    void _pointsUpdated();
+    void _onColorsChanged();
+    void _onHistogramChanged();
     void _onTransferFunctionChanged();
+    void _onScaleChanged( int state );
 
 private:
 
     void _publishTransferFunction();
     void _onTransferFunction();
+    void _onHistogram();
+    void _setGradientStops();
+    void _setHistogram();
+    void _widgetsUpdated();
 
     livre::TransferFunction1D _lut;
```

Comment:
not your code, but _lut is not an explicit name, I still haven't figured out what it means :-)